File: extract_missing_ro.js
const fs = require('fs');
const files = [
    "lib/visualLab/data/poiExtraNicaraguaCitiesV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaEconomicV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaHistoryV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaLandmarksV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaLifeV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaNatureV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaReliefV2.ts"
];

let allMissing = [];

for (const file of files) {
    const content = fs.readFileSync(file, 'utf-8');
    // Basic regex to find POI objects
    const poiRegex = /{[\s\S]*?id:\s*['"]([^'"]+)['"][\s\S]*?name:\s*{([^}]+)}[\s\S]*?}/g;
    
    let match;
    while ((match = poiRegex.exec(content)) !== null) {
        const fullMatch = match[0];
        const id = match[1];
        const nameBlock = match[2];
        
        let enName = '';
        const enMatch = nameBlock.match(/en:\s*['"]([^'"]+)['"]/);
        if (enMatch) {
            enName = enMatch[1];
        }
        
        // Check if descriptionAdvanced exists and has ro
        const descAdvMatch = fullMatch.match(/descriptionAdvanced:\s*{([^}]+)}/);
        let hasRoDesc = false;
        if (descAdvMatch) {
            if (/ro:\s*['"]/.test(descAdvMatch[1])) {
                hasRoDesc = true;
            }
        }
        
        if (!hasRoDesc) {
            allMissing.push({ id, file, name: enName });
        }
    }
}

console.log(JSON.stringify(allMissing, null, 2));

File: extract_missing_ro.py
import re
import json

files = [
    "lib/visualLab/data/poiExtraNicaraguaCitiesV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaEconomicV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaHistoryV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaLandmarksV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaLifeV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaNatureV2.ts",
    "lib/visualLab/data/poiExtraNicaraguaReliefV2.ts"
]

all_missing = []

for file_path in files:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # split by "  {" to get objects
    parts = re.split(r'\n\s*\{\n\s*id: ', content)
    for part in parts[1:]:
        part = "id: " + part
        
        id_match = re.search(r'id:\s*["\']([^"\']+)["\']', part)
        if not id_match:
            continue
        poi_id = id_match.group(1)
        
        # En name
        en_match = re.search(r'name:\s*\{.*?en:\s*["\']([^"\']+)["\']', part, re.DOTALL)
        en_name = en_match.group(1) if en_match else "Unknown"
        
        # descriptionAdvanced check
        desc_adv_match = re.search(r'descriptionAdvanced:\s*\{(.*?)\}', part, re.DOTALL)
        has_ro_desc = False
        if desc_adv_match:
            # check if ro is non-empty string
            ro_match = re.search(r'ro:\s*["\'](.+?)["\']', desc_adv_match.group(1))
            if ro_match and len(ro_match.group(1).strip()) > 0:
                has_ro_desc = True

        # factsAdvanced check
        facts_adv_match = re.search(r'factsAdvanced:\s*\{(.*?)\}', part, re.DOTALL)
        has_ro_facts = False
        if facts_adv_match:
            # check if ro array has items
            ro_facts_match = re.search(r'ro:\s*\[(.*?)\]', facts_adv_match.group(1), re.DOTALL)
            if ro_facts_match and len(ro_facts_match.group(1).strip()) > 0:
                has_ro_facts = True

        if not has_ro_desc or not has_ro_facts:
            all_missing.append({
                "id": poi_id,
                "file": file_path,
                "name": en_name
            })

print(json.dumps(all_missing, indent=2))
